packages/api/endpoints/rules.js

```diff
@@ -137,48 +141,38 @@ async function put(req, res) {
   const { params: { name }, body } = req;
 
   const apiRule = { ...body };
-  let newRule;
+  let translatedRule;
 
   if (name !== apiRule.name) {
     return res.boom.badRequest(`Expected rule name to be '${name}', but found`
       + ` '${body.name}' in payload`);
   }
 
   try {
-    const oldRule = await ruleModel.get({ name });
+    const oldRule = await rulePgModel.get(knex, { name });
+    const oldApiRule = await translatePostgresRuleToApiRule(oldRule, knex);
 
     apiRule.updatedAt = Date.now();
-    apiRule.createdAt = oldRule.createdAt;
+    apiRule.createdAt = oldRule.created_at;
+    const apiPgRule = await translateApiRuleToPostgresRuleRaw(apiRule, knex);
 
     // If rule type is onetime no change is allowed unless it is a rerun
     if (apiRule.action === 'rerun') {
-      return models.Rule.invoke(oldRule).then(() => res.send(oldRule));
+      return invokeRerun(oldApiRule).then(() => res.send(oldApiRule));
     }
 
-    const fieldsToDelete = Object.keys(oldRule).filter(
-      (key) => !(key in apiRule) && key !== 'createdAt'
-    );
-    const postgresRule = await translateApiRuleToPostgresRule(apiRule, knex);
+    const ruleWithTrigger = await updateRuleTrigger(oldRule, apiPgRule, knex);
```

Comment:
What about this refactor:

```
    // If rule type is onetime no change is allowed unless it is a rerun
    if (apiRule.action === 'rerun') {
      return invokeRerun(oldApiRule).then(() => res.send(oldApiRule));
    }

    const apiRuleWithTrigger = await updateRuleTrigger(oldApiRule, apiRule, knex);
    const apiPgRule = await translateApiRuleToPostgresRuleRaw(apiRuleWithTrigger, knex);
```

The key change is moving the call to `translateApiRuleToPostgresRuleRaw`up here. The real benefit of these changes would be downstream in `createRuleTrigger` and `updateRuleTrigger`, which would now handle an API style rule instead of a PG rule. It seems like this refactor would eliminate the need to use `translatePostgresRuleToApiRule` before calling `buildPayload`? 

In essence, the only translation would live in the endpoint, since are primarily writing to PG but returning "API" style records. I think it would reduce the overall number of translation calls/DB queries. And perhaps keep the API helper code simpler to always handle an API rule?

Reply:
I think that's reasonable. I updated `createRuleTrigger` and `updateRuleTrigger` to handle an API style rule and also updated the put endpoint 

[`68155b3` (#2871)](https://github.com/nasa/cumulus/pull/2871/commits/68155b3331587a648cfa8a71f3b863c9bfd173fb)